packages/CodingStandard/src/Fixer/DependencyInjection/InjectToConstructorInjectionFixer.php

```diff
@@ -0,0 +1,239 @@
+            [
+                new CodeSample(
+'<?php
+class SomeClass
+{
+    /**
+     * @inject
+     * @var stdClass
+     */
+    public $property;
+}'),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_DOC_COMMENT, T_PUBLIC, T_VARIABLE]);
+    }
+
+    public function fix(SplFileInfo $file, Tokens $tokens): void
+    {
+        // 1. find annotation @inject
+        foreach ($tokens as $index => $token) {
+            if (! $token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $doc = new DocBlock($token->getContent());
+            $injectAnnotations = $doc->getAnnotationsOfType('inject');
+            if (! count($injectAnnotations)) {
+                continue;
+            }
+
+            // 1. remove it
+            foreach ($injectAnnotations as $injectAnnotation) {
+                $injectAnnotation->remove();
+            }
+
+            $token->setContent($doc->getContent());
+
+            // 2. make public property private
+            for ($i = $index;; ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_PUBLIC)) {
+                    $token->override([T_PRIVATE, 'private']);
+                    break;
+                }
+            }
+
+            // 3. add dependency to constructor
+            $propertyName = '';
+            for ($i = $index;; ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_VARIABLE)) {
+                    $propertyName = ltrim($token->getContent(), '$');
+                    break;
+                }
+            }
+            $varAnnotation = $doc->getAnnotationsOfType('var')[0];
+            $propertyType = $varAnnotation->getTypes()[0];
+
+            // detect constructor
+            $constructMethodPosition = null;
+            for ($i = $index; $i < count($tokens); ++$i) {
+                $token = $tokens[$i];
+                if ($token->isGivenKind(T_FUNCTION)) {
+                    $namePosition = $tokens->getNextNonWhitespace($i);
+                    $methodNameToken = $tokens[$namePosition];
+                    if ($methodNameToken->getContent() === '__construct') {
+                        $constructMethodPosition = $i;
+                        break;
+                    }
+                }
+            }
+
+            // A. has a constructor?
+            if (is_int($constructMethodPosition)) { // "function" token
+                $this->addPropertyToConstructor($tokens, $propertyType, $propertyName, $constructMethodPosition);
+            } else {
+                // B. doesn't have a constructor?
+                $this->addConstructorMethod($tokens, $propertyType, $propertyName);
+            }
+
+            // run again with new tokens; @todo: can this be done any better to notice new __construct method added
+            // by these tokens?
```

Comment:
my approach would be something like;
```
foreach reverse; index, token in tokens
if token is T_CLASS
    fixClass (index)

fixClass(tokens, classIndex)
    classOpenIndex = tokens->getNextTokenOfKind(classyIndex, array('{'));
    classEndIndex = tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, classOpenIndex);

    // now we can search for class attr. and the constructor within the token range as found above

    candidates = this->findAttributeCandidates(tokens, classOpenIndex, classEndIndex);
    if (!count(candidates) {
       return;
    }

    for candidate in candidates
         candidateNames[] = candidate[name];
         this->fixCandidate(tokens, candidate); // fix both PHPDoc and scoping of attr.
                
    this->fixConstructor(candidateNames); // either fix or insert constructor
```

You already got most of the components so with a bit of moving code around this should be doable.
However, please don't take this as me telling you what to!
There are many ways of fixing and yours suits your use case well :)